### PR TITLE
nixos/droidcam: init

### PR DESCRIPTION
**Pull request by suhr**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/programs/droidcam.nix [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{ lib, pkgs, config, ... }:
+
+with lib;
+
+{
+  options.services.tox-node = {
+    enable = mkEnableOption "DroidCam client";
```
Comment (SuperSandro2000) [inline nixos/modules/programs/droidcam.nix:7]: Please add a description.

Comment (suhr) [reply nixos/modules/programs/droidcam.nix:7]: `mkEnableOption` already provides a description: https://github.com/NixOS/nixpkgs/blob/master/lib/options.nix#L92

File: nixos/modules/programs/droidcam.nix [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{ lib, pkgs, config, ... }:
+
+with lib;
+
+{
+  options.services.tox-node = {
```
Comment (aanderse) [inline nixos/modules/programs/droidcam.nix:6]: typo

**Review by aanderse — COMMENTED** [review]
Have not tested, but generally LGTM :+1:

Should we add a NixOS test to make sure the kernel module stuff is all working (ie. the app doesn't crash)? Maybe? I'm not sure if that provides much value or not...

File: nixos/modules/programs/droidcam.nix [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{ lib, pkgs, config, ... }:
+
+with lib;
+
+{
+  options.services.droidcam = {
```
Comment (bachp) [inline nixos/modules/programs/droidcam.nix:6]:
```suggestion
  options.programs.droidcam = {
```

I think this fits better under `programs`

File: nixos/modules/programs/droidcam.nix [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    enable = mkEnableOption "DroidCam client";
+  };
+
+  config = lib.mkIf cfg.enable {
```
Comment (bachp) [inline nixos/modules/programs/droidcam.nix:10]:
```suggestion
  config = lib.mkIf config.programs.droidcam.enable {
```
`cfg` is not defined.

**Review by bachp — CHANGES_REQUESTED** [review]
The module workes after I applied the requested changes.

**Review by bachp — APPROVED** [review]

**Comment by bachp** [comment]
@suhr please squash the commits together

**Pull request merged**